lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/TaskManager.cs

```diff
@@ -200,13 +200,13 @@ internal void Reset()
 
         /// <summary>
         /// This method is called when receiving ICompletedTask event during task running or system shutting down.
-        /// Removes the task from running tasks
+        /// Removes the task from running tasks if it was running
         /// Changes the task state from RunningTask to CompletedTask
         /// </summary>
         /// <param name="completedTask"></param>
         internal void RecordCompletedTask(ICompletedTask completedTask)
         {
-            RemoveRunningTask(completedTask.Id);
+            _runningTasks.Remove(completedTask.Id);
```

Comment:
Maybe it needs a better name then? Right now, we have two ways to remove a taskid from the list of known running tasks, and they are confusingly similar.